build/checkout-secrets.ps1

```diff
@@ -0,0 +1,75 @@
+param (
+  [string]$connectionString = ""
+)
+
+function AcquireLease($blob) {
+  try {
+    return $blob.ICloudBlob.AcquireLease($null, $null, $null, $null, $null)
+  } catch {
+    Write-Host "  Error: $_"
+    return $null
+  } 
+}
+
+# use this for tracking metadata in lease blobs
+$buildName = "3.0." + $env:buildNumber + "_" + $env:SYSTEM_JOBDISPLAYNAME
+
+$azVersion = "1.11.0"
+Import-Module Az.Storage
+$azModule = Get-Module -Name Az.Storage
+if ($azModule.Version -ne $azVersion) {
+  throw "Az.Storage module version $azVersion was not found. Current version: $($azModule.Version)"
+}
+
+# get a blob lease to prevent test overlap
+$storageContext = New-AzStorageContext -ConnectionString $connectionString
+
+While($true) {
```

Comment:
Wondering if we should have a timeout.

Reply:
Yeah, exactly. And we have a way to time it out manually should we need to (by canceling the run). If we see this happening too much we can revisit pretty easily -- or just add more services so we don't have to wait as long.